pkg/services/signingkeys/signingkeysimpl/service.go

```diff
@@ -237,8 +238,12 @@ func (s *Service) decodePrivateKey(ctx context.Context, privateKey []byte) (cryp
 		return nil, errors.New("private key is empty")
 	}
 
-	payload := make([]byte, base64.StdEncoding.DecodedLen(len(privateKey)))
-	_, err := base64.StdEncoding.Decode(payload, privateKey)
+	// Backwards compatibility with old base64 encoding
+	// Can be removed in the future
+	privateKey = bytes.TrimRight(privateKey, "=")
```

Comment:
I don't really understand why you trim `=`s from the encoded form. The `=`s are part of the base64 encoded form and they are necessary to correctly decode.

Here is an example: https://go.dev/play/p/BOy4HsxmnX6. As you see, if you remove the ending `=` then the decoded string is not the original one.

Reply:
It's because you are not using `RawStdEncoding`: https://go.dev/play/p/LZhbtE034EU

Reply:
Ops, I missed that, thanks for clarifying.